Documentation/gettingstarted/host-firewall.rst

```diff
@@ -0,0 +1,156 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _host_firewall:
+
+********************
+Host Firewall (beta)
+********************
+
+This document serves as an introduction to Cilium's host firewall, to enforce
+security policies for Kubernetes nodes.
+
+.. include:: ../beta.rst
+
+Enable the Host Firewall in Cilium
+==================================
+
+.. include:: k8s-install-download-release.rst
+
+Deploy Cilium release via Helm:
+
+.. parsed-literal ::
+
+    helm install cilium |CHART_RELEASE|        \\
+      --namespace kube-system                  \\
+      --set global.kubeProxyReplacement=strict \\
+      --set global.hostFirewall=true           \\
+      --set global.devices=ethX,ethY
+
+At this point, the Cilium-managed nodes are ready to enforce network policies.
+
+
+Enable Policy Audit Mode for the Host Endpoint
+==============================================
+
+When applying `HostPolicies` carelessly, it's easy to block the access to the
+nodes or to break the cluster's normal behavior (ex. by blocking communication
+with kube-api).
+
+To avoid such issues, we can switch the host firewall in audit mode, to
+validate the impact of host policies before enforcing them.
+
+::
+
+    $ HOST_EP_ID=$(cilium endpoint list -o json | jq '.[] | select( .status.identity.id == 1 ).id')
```

Comment:
FWIW `jq` is not available in the Cilium container so this is not necessarily going to 'just work' for the average user.

Reply:
Ok, I think that's a big enough problem that we should find a way to address it. I want users to be able to copy paste those commands as much as possible.

One idea is to restrict the whole guide to a single node. I would ask users to set a label on a node and I'd apply the policy only for that label. Then, we can `jq` the Cilium pod for that node and use it in subsequent commands. So something like:
```bash
export NODE_NAME=k8s1
kubectl label node $NODE_NAME access=ssh
[...]
CILIUM_POD_NAME=$(kubectl -n $CILIUM_NAMESPACE get pods -o json | jq -r '.items[] | select( .metadata.labels."k8s-app" == "cilium" and .spec.nodeName == env.NODE_NAME ).metadata.name')
HOST_EP_ID=$(kubectl -n $CILIUM_NAMESPACE exec $CILIUM_POD_NAME -- cilium endpoint list -o json | jq '.[] | select( .status.identity.id == 1 ).id')
[...]
```
One downside is that it could be a bit harder for users to understand what's happening, as in what info. is `jq` grabbing. So I at least need to be cautious with variable names I use :-)

What do you think?

Reply:
Can't you simply replace `-o json | jq` by `-o jsonpath` with the relevant filter to get the endpoint id?

Reply:
Aaah, that's what `-o jsonpath` is!